coretex/cli/resources/update_script_template.sh

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
+
+# Define Docker-related variables
+DOCKER_PATH={dockerPath}
+REPOSITORY={repository}
+TAG={tag}
+IMAGE={repository}:{tag}
+SERVER_URL={serverUrl}
+STORAGE_PATH={storagePath}
+NODE_ACCESS_TOKEN={nodeAccessToken}
+CONTAINER_NAME={containerName}
+NETWORK_NAME={networkName}
+RESTART_POLICY={restartPolicy}
+PORTS={ports}
+CAP_ADD={capAdd}
+RAM_MEMORY={ramMemory}G
+SWAP_MEMORY={swapMemory}G
+SHARED_MEMORY={sharedMemory}G
+
+# Main execution
+NODE_UPDATE_COMMAND="update_node"
+
+NODE_STATUS_ENDPOINT="http://localhost:21000/status"
+
+# Dump logs to CLI directory
+OUTPUT_DIR="$HOME/.config/coretex"
+mkdir -p "$OUTPUT_DIR"
+
+# Generate the output filename based on the current Unix timestamp
+OUTPUT_FILE="$OUTPUT_DIR/ctx_autoupdate.log"
+
+# Redirect all output to the file
+exec >>"$OUTPUT_FILE" 2>&1
+
+function should_update {{
+    local api_response=$(curl -s "$NODE_STATUS_ENDPOINT")
+    local status=$(echo "$api_response" | sed -n 's/.*"status":\([^,}}]*\).*/\1/p')
+
+    if [ "$status" -eq 2 ]; then
+        echo "Checking node version..."
+        # Get latest image digest from docker hub
+        manifest_output=$($DOCKER_PATH manifest inspect $REPOSITORY:$TAG --verbose)
+
+        # get digest from docker hub
+        digest_line=$(echo "$manifest_output" | grep -o '"digest": ".*"' | head -n 1)
+        latest_digest=$(echo "$digest_line" | cut -d '"' -f 4)
+
+        # get digest from local container 
+        current_digest=$($DOCKER_PATH image ls --digests | grep $REPOSITORY | grep $TAG | awk '{{print $3}}')
+        current_digest=$(echo "$current_digest" | awk '{{$1=$1;print}}')
+
+        # Compare digests
+        if [[ $latest_digest != $current_digest ]]; then
+            return 1 # Return False since there is new image to be pulled from docker hub
+        else
+            return 1
+            # echo "Node version is up to date."
+            # exit 0
+        fi
+    else
+        echo "Node is not active."
+        exit 0
+    fi
+}}
+
+function pull_image {{
+    if $DOCKER_PATH image pull "$IMAGE" >/dev/null; then
```

Comment:
Can `>/dev/null` be removed? Output is already getting redirected into a log file.